Create method SearchService.deleteResources() to avoid overhead of calling deleteResource() one document at a time.

labkey-matthewb opened the pull request:
#### Rationale
truncateSampleSet() is very slow, this reduces the cost due to updating the full-text index. 

#### Changes
* move the delete document work to the background indexing thread 
* avoid a lot of IndexManager.deleteDocument() overhead (calling maybeRefresh() for each document)

#### Related
https://github.com/LabKey/platform/pull/1286
https://github.com/LabKey/testAutomation/pull/375

labkey-matthewb committed:
Create method SearchService.deleteResources() to avoid overhead of calling deleteResource() one document at a time.

labkey-adam commented on experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java:
```diff
@@ -17,6 +17,7 @@
 package org.labkey.experiment.api;
 
 import com.google.common.collect.Iterables;
+import com.google.gwt.user.client.Command;
```
Remove unused import

labkey-adam reviewed (changes requested):
Just remove the unused import

labkey-adam committed:
Issue 40601: Too many calls to WritableIndexManagerImpl.maybeRefresh()

Throttle maybeRefresh() calls to once every five seconds

labkey-adam committed:
Issue 40601: Too many calls to WritableIndexManagerImpl.maybeRefresh()

Throttle maybeRefresh() calls using Quartz timer to ensure method is called within five seconds of every index modification

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_deletedocuments

labkey-matthewb committed:
commit() calls maybeRefresh() directly

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_deletedocuments

labkey-matthewb committed:
Add refreshNow() to WaitForIndexerAction.
Replace container path with container guid in the document ids for webdav resources.

labkey-matthewb committed:
use TaskListener to implement WaitForIndexerAction

labkey-matthewb committed:
Merge remote-tracking branch 'origin/develop' into fb_deletedocuments

labkey-matthewb committed:
use TaskListener to implement WaitForIndexerAction
